### ISSUE TITLE
I have two questions

### ISSUE THREAD

**Issue by goonm** [frankkramer-lab/MIScnn#2]
Thank you for sharing your code.
I have two questions.

1. What is the main difference between 'patchwise-crop' and 'patchwise-grid'? (Preprocessor analysis)

2. I know that 'residual block' uses not concatenate but add.
(On the contrary, 'dense block' uses not add but concatenate)
But "residual.py"(Residual 3D-Unet) performed concatenate in the 'residual block'.
Please explain more about this.

**Comment by muellerdo** [comment]
Hey goonm,

sorry for the late response.

> What is the main difference between 'patchwise-crop' and 'patchwise-grid'? (Preprocessor analysis)

The patchwise-crop modus creates patches by randomly cropping a patch out of the image. In each patch creation call, a new randomly cropped patch will be created. This is the default and preferred modus.   

The patchwise-grid modus splits the image into a list of patches according to a grid. This modus uses always the same patches in each patch creation call. This result into way faster training. But even with shuffling and extensive data augmentation, this modus can result into overfitting on your training data.  

It is recommended to use the "on-the-fly" patch creation by random cropping -> patchwise-crop modus

> I know that 'residual block' uses not concatenate but add

I based the code for the residual and dense U-Net version on the following publication:
Optimized High Resolution 3D Dense-U-Net Network for Brain and Spine Segmentation, Kolařík, M., Burget, R., Uher, V., Říha, K., & Dutta,  M. K. (2019)

When looking on the original residual paper (https://arxiv.org/pdf/1711.10684.pdf) or the U-Net Residual paper (https://arxiv.org/pdf/1711.10684.pdf), you are right. Theoretically, the residual U-Net block should have an add layer. Still, it wouldn't be a dense block, because you will need a concatenation after each convolution (https://arxiv.org/pdf/1608.06993.pdf).

I will have a look on it, and probably change the concatenation in the residual block to an add layer. I will also add a "modified"/intermediate residual/dense U-Net version using only a single concatenation in each block (current MIScnn residual implementation).

I'm planing to do a comparison between several U-Net variants. It will be quite interesting to compare the performance between residual, intermediate and dense U-Net on a few biomedical data sets.

Greetings,
Dominik

**Comment by muellerdo** [comment]
Fixed U-Net Residual Architecture.
Added U-Net Compact Architecture.
